Makefile

```diff
@@ -472,6 +472,19 @@ endif
 
 default: finish
 
+# TODO: specify SDK version, e.g. libc.so.wasi-sdk-21, as SO_NAME once `wasm-ld` supports it
+$(SYSROOT_LIB)/libc.so: $(SYSROOT_LIB)/libc.so.a $(BUILTINS_LIB)
+	$(CC) -nostdlib -shared -o $@ -Wl,--whole-archive $< -Wl,--no-whole-archive $(BUILTINS_LIB)
+
+$(SYSROOT_LIB)/libc.so.a: \
+		$(filter-out $(MUSL_PRINTSCAN_OBJS),$(LIBC_OBJS)) \
+		$(MUSL_PRINTSCAN_LONG_DOUBLE_OBJS) \
+		$(LIBWASI_EMULATED_MMAN_OBJS) \
+		$(LIBWASI_EMULATED_PROCESS_CLOCKS_OBJS) \
+		$(LIBWASI_EMULATED_GETPID_OBJS) \
+		$(LIBWASI_EMULATED_SIGNAL_OBJS) \
+		$(LIBWASI_EMULATED_SIGNAL_MUSL_OBJS)
```

Comment:
including EMULATED things in libc is an unrelated change. please drop.

Reply:
I want to make sure application programmers can use these features in a shared library context if they need them.  Putting them in libc.so would be the most convenient option, but I'm open to alternatives.  Is your main concern that they make libc.so larger that it otherwise needs to be?  If so, would you prefer that they go in a separate library, e.g. libc-emulated.so?

Reply:
> I want to make sure application programmers can use these features in a shared library context if they need them. Putting them in libc.so would be the most convenient option, but I'm open to alternatives. Is your main concern that they make libc.so larger that it otherwise needs to be? If so, would you prefer that they go in a separate library, e.g. libwasi-emulated.so?

sort of.
i feel it's better to keep things consistent between .a and .so where reasonable.
yes, i prefer to have libwasi-emulated-process-clocks.so etc.